src/app_service/service/message/service.nim

```diff
@@ -222,6 +227,33 @@ QtObject:
       reactionId: r.id, reactionFrom: r.`from`)
       self.events.emit(SIGNAL_MESSAGE_REACTION_FROM_OTHERS, data)
 
+  proc handleMessagesReload(self: Service, communityId: string) =
+    var keys = newSeq[string]()
+    for k in self.msgCursor.keys:
+      keys.add(k)
+    for k in keys:
+      self.msgCursor.del(k)
+
+    keys = newSeq[string]()
```

Comment:
@PascalPrecht instead of this you could use `keys = @[]` that will clear the sequence.